engine/runtime-fat-jar/src/main/java/module-info.java

```diff
@@ -3,6 +3,11 @@
   requires java.net.http;
   // Because of akka.util.Unsafe
   requires jdk.unsupported;
+  requires org.enso.syntax;
+  // profiling-utils is not needed for runtime. It is needed for runner.jar,
+  // but we have to include it here so that the forwarding in IsolatedClassLoader
+  // works.
+  requires org.enso.profiling;
```

Comment:
Removal of these unnecessary dependencies tracked in https://github.com/enso-org/enso/issues/10157